Unable to launch simulator because of Java.lang.StringIndexOutOfBoundsException in latest as of reporting (7.0.236)

**Describe the bug**
Java.lang.StringIndexOutOfBoundsException in latest as of reporting (7.0.236) when trying to run the simulator. 
The project has been downloaded from codenameone initilizr with the following settings

<img width="1542" height="434" alt="Image" src="https://github.com/user-attachments/assets/e8504854-2074-4733-93bc-ab6db822648d" />
Intelij Version: 

<img width="545" height="133" alt="Image" src="https://github.com/user-attachments/assets/12a896d1-7db9-470c-9f3e-17747d9076f5" />

**Expected behavior**
Expect simulator to launch
**Screenshots**
`Failed to execute goal com.codenameone:codenameone-maven-plugin:7.0.236:css (cn1-process-classes) on project myappname-common: An error occurred while compiling the CSS files.  Inputs: /home/<Path>/myappname (1)/common/src/main/css/theme.css, output: /home/<Path>/myappname (1)/common/target/classes/theme.res, merge file: /home/<Path>/myappname (1)/common/target/css/theme.css -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.codenameone:codenameone-maven-plugin:7.0.236:css (cn1-process-classes) on project myappname-common: An error occurred while compiling the CSS files.  Inputs: /home/<Path>/myappname (1)/common/src/main/css/theme.css, output: /home/<Path>/myappname (1)/common/target/classes/theme.res, merge file: /home/<Path>/myappname (1)/common/target/css/theme.css
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: An error occurred while compiling the CSS files.  Inputs: /home/<Path>/myappname (1)/common/src/main/css/theme.css, output: /home/<Path>/myappname (1)/common/target/classes/theme.res, merge file: /home/<Path>/myappname (1)/common/target/css/theme.css
    at com.codename1.maven.CompileCSSMojo.executeImpl (CompileCSSMojo.java:265)
    at com.codename1.maven.CompileCSSMojo.executeImpl (CompileCSSMojo.java:55)
    at com.codename1.maven.AbstractCN1Mojo.execute (AbstractCN1Mojo.java:195)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :myappname-common
`
 
**Additional context**
I have tried both java 8,17,and 21 (setting language level to 8) to check if the issue is with jdk but am getting same error for all the java versions

--- a/Ports/JavaSE/src/com/codename1/impl/javase/JavaSEPort.java
+++ b/Ports/JavaSE/src/com/codename1/impl/javase/JavaSEPort.java
@@ -14337,8 +14337,18 @@ public class JavaSEPort extends CodenameOneImplementation {
         autoLocalizationBundle = null;
     }
 
-    private File findDefaultLocalizationBundleFile() {
-        File localizationDir = findLocalizationDirectory();
+    File findDefaultLocalizationBundleFile() {
+        return findDefaultLocalizationBundleFile(getCWD());
+    }
+
+    // Returns null when no real bundle file exists. Previously this method
+    // returned a non-existent `Bundle.properties` path as a fallback, which
+    // combined with `findLocalizationDirectory`'s old mkdirs() to create empty
+    // ghost bundles in modules the developer never asked to localize. Now the
+    // auto-bundle activates only when the developer ships at least one
+    // bundle file, matching the project-level opt-in semantics.
+    static File findDefaultLocalizationBundleFile(File projectDir) {
+        File localizationDir = findLocalizationDirectory(projectDir);
         if (localizationDir == null) {
             return null;
         }
@@ -14372,10 +14382,10 @@ public class JavaSEPort extends CodenameOneImplementation {
             java.util.Collections.sort(bundles);
             return bundles.get(0);
         }
-        return preferred;
+        return null;
     }
 
-    private void collectLocalizationBundles(File dir, java.util.List<File> out) {
+    private static void collectLocalizationBundles(File dir, java.util.List<File> out) {
         File[] files = dir.listFiles();
         if (files == null) {
             return;
@@ -14389,24 +14399,48 @@ public class JavaSEPort extends CodenameOneImplementation {
         }
     }
 
-    private File findLocalizationDirectory() {
-        File projectDir = getCWD();
-        File[] candidates = new File[]{
-            new File(projectDir, "src" + File.separator + "main" + File.separator + "l10n"),
-            new File(projectDir, "l10n"),
-            new File(projectDir, "src" + File.separator + "l10n")
-        };
+    File findLocalizationDirectory() {
+        return findLocalizationDirectory(getCWD());
+    }
+
+    // Resolves the project's localization bundle directory for the auto-bundle.
+    //
+    // The simulator forks `cn1:run` from the `javase/` module, so cwd is `javase/`
+    // -- but the developer's bundles live in the sibling `common/` module under
+    // `common/src/main/l10n`. Issue #4850: previous versions of this method
+    // looked only at cwd, missed the real bundle, and then auto-created a
+    // throwaway `javase/src/main/l10n/Bundle.properties` via mkdirs(). The
+    // throwaway file accumulated wormhole-poisoned `@im=@im` entries from older
+    // CN1 versions; even after users cleaned the *real* bundle in `common/`,
+    // every CN1CSSCLI subprocess respawn loaded the ghost bundle from `javase/`,
+    // crashed inside `parseTextFieldInputMode`, and CSSWatcher restarted it in
+    // an infinite respawn loop.
+    //
+    // New rules:
+    //  1. Check the current module first (cwd/src/main/{l10n,i18n}).
+    //  2. Then check the sibling `common/` module (matches CN1 maven layout
+    //     and CSSWatcher.addLocalizationCandidates).
+    //  3. Never auto-create the directory. Project-level opt-in: if the
+    //     developer hasn't set up localization, the auto-bundle is a no-op.
+    static File findLocalizationDirectory(File projectDir) {
+        if (projectDir == null) {
+            return null;
+        }
+        java.util.List<File> candidates = new java.util.ArrayList<File>();
+        candidates.add(new File(projectDir, "src" + File.separator + "main" + File.separator + "l10n"));
+        candidates.add(new File(projectDir, "src" + File.separator + "main" + File.separator + "i18n"));
+        candidates.add(new File(projectDir, "l10n"));
+        candidates.add(new File(projectDir, "src" + File.separator + "l10n"));
+        File parent = projectDir.getParentFile();
+        if (parent != null) {
+            File commonModule = new File(parent, "common");
+            candidates.add(new File(commonModule, "src" + File.separator + "main" + File.separator + "l10n"));
+            candidates.add(new File(commonModule, "src" + File.separator + "main" + File.separator + "i18n"));
+        }
         for (File dir : candidates) {
-            if (dir.exists() && dir.isDirectory()) {
+            if (dir != null && dir.exists() && dir.isDirectory()) {
                 return dir;
             }
-        }
-        File fallback = candidates[0];
-        if (!fallback.exists()) {
-            fallback.mkdirs();
-        }
-        if (fallback.exists() && fallback.isDirectory()) {
-            return fallback;
         }
         return null;
     }

--- a/tests/core/test/com/codename1/impl/javase/AutoLocalizationBundleTest.java
+++ b/tests/core/test/com/codename1/impl/javase/AutoLocalizationBundleTest.java
@@ -7,6 +7,7 @@ import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.OutputStream;
 import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Properties;
@@ -82,6 +83,9 @@ public class AutoLocalizationBundleTest extends AbstractTest {
 
             verifySetBundleSmokeOnFreshProject(ctor, tempDir);
             verifySetBundleHealsLegacyWormholeFile(ctor, tempDir);
+            verifyFindLocalizationDirectoryDoesNotAutoCreate(tempDir);
+            verifyFindLocalizationDirectoryWalksToCommonSibling(tempDir);
+            verifyFindDefaultBundleReturnsNullWhenNoBundleFile(tempDir);
 
             return true;
         } finally {
@@ -184,6 +188,85 @@ public class AutoLocalizationBundleTest extends AbstractTest {
         } finally {
             manager.setBundle(savedBundle);
         }
+    }
+
+    /// Issue #4850 root cause: `findLocalizationDirectory` used to call
+    /// `mkdirs()` on `<cwd>/src/main/l10n` whenever the directory was missing,
+    /// then `findDefaultLocalizationBundleFile` returned a non-existent
+    /// `Bundle.properties` path as a fallback. The CSS subprocess inherits cwd
+    /// = `javase/`, so this auto-created a ghost bundle in the wrong module
+    /// that older CN1 versions then poisoned with `@im=@im`. After this fix,
+    /// no l10n dir on disk = no auto-bundle (project-level opt-in).
+    private void verifyFindLocalizationDirectoryDoesNotAutoCreate(File tempDir) throws Exception {
+        File freshModule = new File(tempDir, "no-l10n-module");
+        if (!freshModule.mkdirs()) {
+            throw new RuntimeException("Failed to create test module dir " + freshModule);
+        }
+
+        Method findLocDir = Class.forName("com.codename1.impl.javase.JavaSEPort")
+                .getDeclaredMethod("findLocalizationDirectory", File.class);
+        findLocDir.setAccessible(true);
+
+        Object result = findLocDir.invoke(null, freshModule);
+        assertNull(result, "findLocalizationDirectory must return null when no l10n dir exists");
+
+        File ghostDir = new File(freshModule, "src" + File.separator + "main" + File.separator + "l10n");
+        assertBool(!ghostDir.exists(), "findLocalizationDirectory must not auto-create src/main/l10n");
+    }
+
+    /// Issue #4850: the simulator forks `cn1:run` from `javase/` while the
+    /// developer's bundles live in the sibling `common/` module. The new
+    /// `findLocalizationDirectory` walks up to find `../common/src/main/l10n`
+    /// when the current module has no l10n dir of its own, mirroring
+    /// `CSSWatcher.addLocalizationCandidates`.
+    private void verifyFindLocalizationDirectoryWalksToCommonSibling(File tempDir) throws Exception {
+        File rootProject = new File(tempDir, "multi-module-project");
+        File javaseModule = new File(rootProject, "javase");
+        File commonL10n = new File(rootProject, "common" + File.separator + "src" + File.separator + "main" + File.separator + "l10n");
+        if (!javaseModule.mkdirs() || !commonL10n.mkdirs()) {
+            throw new RuntimeException("Failed to create multi-module project layout under " + rootProject);
+        }
+
+        Method findLocDir = Class.forName("com.codename1.impl.javase.JavaSEPort")
+                .getDeclaredMethod("findLocalizationDirectory", File.class);
+        findLocDir.setAccessible(true);
+
+        File result = (File) findLocDir.invoke(null, javaseModule);
+        assertNotNull(result, "findLocalizationDirectory must locate sibling common/src/main/l10n");
+        assertEqual(commonL10n.getCanonicalPath(), result.getCanonicalPath(),
+                "findLocalizationDirectory should resolve to the common module's l10n dir when cwd is javase");
+
+        // Local module wins when both exist.
+        File javaseL10n = new File(javaseModule, "src" + File.separator + "main" + File.separator + "l10n");
+        if (!javaseL10n.mkdirs()) {
+            throw new RuntimeException("Failed to create local l10n dir " + javaseL10n);
+        }
+        File preferLocal = (File) findLocDir.invoke(null, javaseModule);
+        assertEqual(javaseL10n.getCanonicalPath(), preferLocal.getCanonicalPath(),
+                "Local module's l10n dir should take precedence over common");
+    }
+
+    /// `findDefaultLocalizationBundleFile` previously returned a non-existent
+    /// `Bundle.properties` path when the dir was empty; that triggered
+    /// `AutoLocalizationBundle.persist()` to create the empty file even when
+    /// the project shipped no bundles. Now it returns null and
+    /// `enableAutoLocalizationBundle` no-ops.
+    private void verifyFindDefaultBundleReturnsNullWhenNoBundleFile(File tempDir) throws Exception {
+        File emptyL10nModule = new File(tempDir, "empty-l10n-module");
+        File emptyL10n = new File(emptyL10nModule, "src" + File.separator + "main" + File.separator + "l10n");
+        if (!emptyL10n.mkdirs()) {
+            throw new RuntimeException("Failed to create empty l10n dir " + emptyL10n);
+        }
+
+        Method findDefaultBundle = Class.forName("com.codename1.impl.javase.JavaSEPort")
+                .getDeclaredMethod("findDefaultLocalizationBundleFile", File.class);
+        findDefaultBundle.setAccessible(true);
+
+        Object result = findDefaultBundle.invoke(null, emptyL10nModule);
+        assertNull(result, "findDefaultLocalizationBundleFile must return null when the l10n dir has no .properties files");
+
+        File preferred = new File(emptyL10n, "Bundle.properties");
+        assertBool(!preferred.exists(), "findDefaultLocalizationBundleFile must not create Bundle.properties");
     }
 
     private Properties load(File file) throws Exception {

